### DISCUSSION TITLE
[Bug]: emojis sent via JSON messages to the public broker are lost

### DISCUSSION THREAD

**Discussion by ondras** [meshtastic/firmware#5051] [Troubleshooting]
### Category

Other

### Hardware

Heltec V3

### Firmware Version

v2.5.5.e182ae7

### Description

This might not be the proper repository to file an issue against. I was unable to find a better one.

Nodeinfo JSON messages sent to a public MQTT broker are somewhat broken: their `longname` fields have their emojis replaced with `\u0000`.

Apparently, the public MQTT broker has issues with emojis (not sure which codepoints in particular -- perhaps those outside of [BMP](https://en.wikipedia.org/wiki/Plane_(Unicode)#Basic_Multilingual_Plane)?). My reproducible example using `mosquitto_sub` and `mosquitto_pub`:

```
// publisher
$ mosquitto_pub -t "msh/test" -h mqtt.meshtastic.org -u meshdev -P large4cats -i test2 -m "hi"
$ mosquitto_pub -t "msh/test" -h mqtt.meshtastic.org -u meshdev -P large4cats -i test2 -m "hi 💩"


// subscriber
$ mosquitto_sub -t "msh/test" -h mqtt.meshtastic.org -u meshdev -P large4cats -i test
hi
```

Note the second missing message. Now compare with my own mosquitto server:

```
// publisher
$ mosquitto_pub -t "msh/test" -h mqtt.home -i test2 -m "hi 💩"


// subscriber
$ mosquitto_sub -t "msh/test" -h mqtt.home -i test
hi 💩
```

**Comment by fifieldt** [comment]
This was likely fixed by:

https://github.com/meshtastic/firmware/pull/6011
